Dockerfile

```diff
@@ -0,0 +1,116 @@
+FROM registry.suse.com/suse/sle15:15.6 AS fetcher
+
+ENV BINS_DIR=/tmp/binaries/darwin
+ENV HELM_VERSION=3.12.2
+ENV KUBECTL_VERSION=1.29.3
+
+RUN mkdir -p ${BINS_DIR}
+RUN curl -s -o helm-v${HELM_VERSION}-darwin-arm64.tar.gz https://get.helm.sh/helm-v${HELM_VERSION}-darwin-arm64.tar.gz \
+    && tar -zxvf helm-v${HELM_VERSION}-darwin-arm64.tar.gz \
+    && mv darwin-arm64/helm ${BINS_DIR}/helm \
+    && curl -LO "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/darwin/arm64/kubectl" \
+    && chmod +x ./kubectl \
+    && mv kubectl ${BINS_DIR}/kubectl
+
+ENV BINS_DIR=/tmp/binaries/linux
+RUN mkdir -p ${BINS_DIR}
+RUN curl -s -o helm-v${HELM_VERSION}-linux-amd64.tar.gz https://get.helm.sh/helm-v${HELM_VERSION}-linux-amd64.tar.gz \
+    && tar -zxvf helm-v${HELM_VERSION}-linux-amd64.tar.gz \
+    && mv linux-amd64/helm ${BINS_DIR}/helm \
+    && curl -LO "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl" \
+    && chmod +x ./kubectl \
+    && mv kubectl ${BINS_DIR}/kubectl
+
+ENV BINS_DIR=/tmp/binaries/windows
+RUN mkdir -p ${BINS_DIR}
+RUN curl -s -o helm-v${HELM_VERSION}-windows-amd64.tar.gz https://get.helm.sh/helm-v${HELM_VERSION}-windows-amd64.tar.gz \
+    && tar -zxvf helm-v${HELM_VERSION}-windows-amd64.tar.gz \
+    && mv windows-amd64/helm.exe ${BINS_DIR}/helm.exe \
+    && curl -LO "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/windows/amd64/kubectl.exe" \
+    && mv kubectl.exe ${BINS_DIR}/kubectl.exe
+
+FROM --platform=$BUILDPLATFORM openapitools/openapi-generator-cli:v6.6.0 AS backend-client-generator
+
+COPY ui /ui
+WORKDIR /ui
+
+RUN /usr/local/bin/docker-entrypoint.sh generate -g typescript-axios -i https://api.apps.rancher.io/api-docs -o autogenerated/client/backend --additional-properties usePromises=true --model-name-suffix=DTO
+
+FROM --platform=$BUILDPLATFORM dp.apps.rancher.io/containers/nodejs:22-dev AS client-builder
+WORKDIR /ui
+# cache packages in layer
+COPY ui/package.json /ui/package.json
+COPY ui/package-lock.json /ui/package-lock.json
+RUN --mount=type=cache,target=/usr/src/app/.npm \
+    npm set cache /usr/src/app/.npm && \
+    npm ci
+# install
+COPY ui /ui
+COPY --from=backend-client-generator /ui/autogenerated /ui/autogenerated
+RUN npm run build
+
+FROM --platform=$BUILDPLATFORM dp.apps.rancher.io/containers/nodejs:22-dev AS backend-builder
+WORKDIR /backend
+# cache packages in layer
+COPY backend/package.json /backend/package.json
+COPY backend/package-lock.json /backend/package-lock.json
+RUN --mount=type=cache,target=/usr/src/app/.npm \
+    npm set cache /usr/src/app/.npm && \
+    npm ci
+# install
+COPY backend /backend
+
+FROM dp.apps.rancher.io/containers/nodejs:22
```

Comment:
@paleloser Are you using a service account to pull content?

Reply:
There's nothing configured when it comes to Docker yet. I'll have to figure it out 🤔

Reply:
I'm going to improve the pipeline to build the imag whenever the `Dockerfile` changes, so we exercise the build, login, etc.